src/pip/_internal/commands/install.py

```diff
@@ -1,3 +1,7 @@
+
+# The following comment should be removed at some point in the future.
+# mypy: strict-optional=False
```

Comment:
nit: It'd probably help to flag/add comments to lines that require this, especially if you looked into this and have some insight into what could be done.